protocol-jackson/src/main/java/io/camunda/zeebe/protocol/jackson/record/AbstractJsonSerializable.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.protocol.jackson.record;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.zeebe.protocol.record.JsonSerializable;
+import java.io.UncheckedIOException;
+
+/**
+ * Base abstract class for all immutable objects in order to implement {@link JsonSerializable} such
+ * that it will not be an attribute, but instead act as a normal method. Unfortunately this cannot
+ * be an interface, as default interface methods act as default values for attributes and not plain
+ * methods.
+ */
+abstract class AbstractJsonSerializable implements JsonSerializable {
+  /** @deprecated use your own {@link ObjectMapper} instead of relying on this one */
+  @Deprecated(since = "1.2.0")
+  private static final ObjectMapper JSON_WRITER = new ObjectMapper();
+
+  /** @deprecated use a {@link ObjectMapper} directly instead of calling this method */
+  @Override
+  @Deprecated(since = "1.2.0")
+  public String toJson() {
```

Comment:
🔧 I think that `JsonSerializable.toJson` should also be deprecated

Reply:
I wasn't yet sure about this, as that's definitely a breaking change as it's on the older module, and I wanted to think about it a bit more (also run it by menski).

Reply:
❓ Deprecation is not a breaking change right? It's simply stating that in the next major version, this method may be removed.

👍 It's good to give this more consideration though